### ISSUE TITLE
`cabal install` succeeds, but `hwk` does not run

### ISSUE THREAD

**Issue by andreasabel** [juhp/hwk#2]
With GHC 9.0 (and cabal-install 3.x):
```
$ cabal install hwk
...
$ echo 1 2 | hwk reverse
ERROR: Won't compile:
<no location info>: error:
    Could not load module ‘Data.List.Extra’
    It is a member of the hidden package ‘extra-1.7.9’.
    You can run ‘:set -package extra’ to expose it.
    (Note: this unloads all the modules in the current scope.)
    It is a member of the hidden package ‘extra-1.7.9’.
    You can run ‘:set -package extra’ to expose it.
    (Note: this unloads all the modules in the current scope.)
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
```
Does installation rely on specific cabal settings?  Maybe then they should be given in the instructions...

**Comment by andreasabel** [comment]
See a similar problem at https://github.com/gelisam/hawk/issues/215.

I can fix the problem with
```
$ cabal install --lib extra
```

**Comment by juhp** [comment]
Sorry for not seeing this/responding earlier.

I think the "workaround" you mentioned (`cabal install --lib ...`) is the only viable one at the moment.
Though I am open to other suggestions of how to handle this better if it's possible.

**Comment by andreasabel** [comment]
Is there some upstream awareness/documentation/migration guide on such issues.  I have the feeling that GHC-9 and v2-cabal have created similar problems in other projects.  At least I come across similar errors quite often these days.

**Comment by juhp** [comment]
Right, it is a good question - I don't know TBH, not aware of anything particular.
I even only really became aware of `cabal install --lib` fairly recently...

So far I only found https://cabal.readthedocs.io/en/3.6/cabal-commands.html#adding-libraries-to-ghc-package-environments.
